kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/DefaultKernelUtils.java

```diff
@@ -74,6 +72,8 @@ public static Type findSubFieldType(GroupType groupType, StructField field) {
         return null;
     }
 
+    // TODO: Move these precondition checks into a separate utility class.
```

Comment:
please keep track of the TODOs and make sure all these are done before the 3.0 release.

Reply:
Agree.